src/strings.rs

```diff
@@ -20,54 +20,26 @@ impl StringTable {
     ///
     /// If the string was already inserted before, the string is deduplicated
     /// and the index to the previous string is returned.
-    pub fn insert<S: Into<String>>(&mut self, s: S) -> u64 {
-        // Horrible news, we need to clone s, even we would not need to:
-        // https://github.com/rust-lang/rust/pull/50821
-        let s = s.into();
-        let s_len = s.len();
-        match self.indexed_data.entry(s) {
-            hash_map::Entry::Occupied(entry) => *entry.into_mut(),
-            hash_map::Entry::Vacant(entry) => {
-                let idx = *entry.insert(self.size_in_bytes);
-                self.size_in_bytes = self.size_in_bytes + s_len as u64 + 1;
-                idx
-            }
+    pub fn insert(&mut self, s: &str) -> u64 {
+        // Horrible news, we cannot use entry API since it does not support Borrow
```

Comment:
Could you please add this tracking ticket: https://github.com/rust-lang/rust/issues/56167